tests/unit/test_client.py

```diff
@@ -0,0 +1,102 @@
+import os.path
+
+from botocore.exceptions import ClientError
+import pytest
+
+from moto import mock_s3
+from dotenv import load_dotenv
+
+import re
+from IIIFingest.client import Client
+
+load_dotenv()
+
+
+@mock_s3
+class TestClient:
+    test_aws_profile = os.getenv('TEST_AWS_PROFILE', default="tester")
+    # must match client params e.g. account space and environment - see conftest.py
+    bucket_name = os.getenv('TEST_BUCKET', default="edu.harvard.huit.lts.mps.test-testing-space-dev")
+    
+    # Define manifest-level metadata
+    manifest_level_metadata = {
+        "labels": ["Test Manifest"],
+        "summary": "A test manifest for ingest",
+        "rights": "http://creativecommons.org/licenses/by-sa/3.0/",
+    }
+
+    def test_client_upload(self, test_images, boto_session, test_client):
+        boto_session.resource('s3').create_bucket(Bucket=self.bucket_name)
+
+        image_path = test_images["27.586.1-cm-2016-02-09.tif"]["filepath"]
+
+        client = test_client
+
+        images = [{"label": "Test Image", "filepath": image_path}]
+        assets = client.upload(images, s3_path="testing")
+
+        assert assets is not None
+        assert assets[0].filepath == image_path
```

Comment:
In addition to checking the asset `filepath`, you might also consider checking that the uploaded asset has an `asset_id` since that is created automatically as part of the upload process. If the image has an `id` property, it's supposed to be included in the `asset_id`, so you might consider checking both cases (with and without an image id).